pkg/scheduler/backend/heap/heap.go

```diff
@@ -102,6 +105,9 @@ func (h *data[T]) Push(kv interface{}) {
 
 // Pop is supposed to be called by container/heap.Pop only.
 func (h *data[T]) Pop() interface{} {
+	if len(h.queue) == 0 {
```

Comment:
I think we lack the check for `Pinfo` popped by the queue.
https://github.com/kubernetes/kubernetes/blob/e95441591235032994666ff91cfcd77a13d8c7b2/pkg/scheduler/backend/queue/active_queue.go#L196-L200

For example, we directly add attempts here.

SO, instead of return a nil, maybe let's return an error?

Reply:
Modified `Pop()` method is called only by `container/heap.Pop`, so we can't return anything more from this method, because we have to satisfy an interface. `backend.Heap` exposes a second `Pop()` method that is used within a scheduling queue, which returns an error when we get a nil:

https://github.com/kubernetes/kubernetes/blob/1f157bcb90a4e4908e55109998263bb5c9a0f6d0/pkg/scheduler/backend/heap/heap.go#L176-L186

Reply:
Ah, I mislooked 🤦